File: TelerikUIExamples/TelerikUIExamples/LocalizedCalendar.m
//
//  LocalizedCalendar.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "LocalizedCalendar.h"
#import <TelerikUI/TelerikUI.h>

@interface LocalizedCalendar ()

@property (nonatomic, strong) TKCalendar *calendarView;

@end

@implementation LocalizedCalendar

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
        [self addOption:@"Russian" selector:@selector(selectRussian)];
        [self addOption:@"German" selector:@selector(selectGerman)];
        [self addOption:@"Hebrew" selector:@selector(selectHebrew)];
        [self addOption:@"Chinese" selector:@selector(selectChinese)];
        [self addOption:@"Islamic"  selector:@selector(selectIslamic)];
        
        self.selectedOption = 2;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.calendarView = [[TKCalendar alloc] initWithFrame:self.view.bounds];
    [self.view addSubview:self.calendarView];
    
    [self selectHebrew];
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    self.calendarView.frame = self.view.bounds;
}

- (void)selectRussian
{
    self.calendarView.calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
// >> localization-localeproperty
    self.calendarView.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"ru_RU"];
// << localization-localeproperty    
    [self.calendarView.presenter update:NO];
}

- (void)selectGerman
{
    self.calendarView.calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierGregorian];
    self.calendarView.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"de_DE"];
    [self.calendarView.presenter update:NO];
}

- (void)selectHebrew
{
    self.calendarView.calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierHebrew];
    self.calendarView.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"he_IL"];
    [self.calendarView.presenter update:NO];
}

- (void)selectChinese
{
// >> localization-chinesecalendar
    self.calendarView.calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierChinese];
// << localization-chinesecalendar
    self.calendarView.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"zh_Hans_SG"];
    [self.calendarView.presenter update:NO];
}

- (void)selectIslamic
{
    self.calendarView.calendar = [[NSCalendar alloc] initWithCalendarIdentifier:NSCalendarIdentifierIslamic];
    self.calendarView.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"ar-QA"];
// >> localization-update
    [self.calendarView.presenter update:NO];
// << localization-update
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/LocalizedCalendar.swift
//
//  LocalizedCalendar.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class LocalizedCalendar: TKExamplesExampleViewController {
    
    let calendarView = TKCalendar()
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Russian", action: selectRussian)
        self.addOption("German", action: selectGerman)
        self.addOption("Hebrew", action: selectHebrew)
        self.addOption("Chinese", action: selectChinese)
        self.addOption("Islamic", action: selectIslamic)
        
        self.selectedOption = 2
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.calendarView.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(self.calendarView)
        
        self.selectHebrew()
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        self.calendarView.frame = self.view.bounds
    }
    
    func selectRussian() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
        self.calendarView.locale = NSLocale(localeIdentifier: "ru_RU")
        self.calendarView.presenter.update(false)
    }
    
    func selectGerman() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
        self.calendarView.locale = NSLocale(localeIdentifier: "de_DE")
        self.calendarView.presenter.update(false)
    }
    
    func selectHebrew() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierHebrew)!
        self.calendarView.locale = NSLocale(localeIdentifier: "he_IL")
        self.calendarView.presenter.update(false)
    }
    
    func selectChinese() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierChinese)!
        self.calendarView.locale = NSLocale(localeIdentifier: "zh_Hans_SG")
        self.calendarView.presenter.update(false)
    }
    
    func selectIslamic() {
        self.calendarView.calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierIslamic)!
        self.calendarView.locale = NSLocale(localeIdentifier:"ar-QA")
        self.calendarView.presenter.update(false)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
